google-cloud-storage/src/main/java/com/google/cloud/storage/OpenTelemetryBootstrappingUtils.java

```diff
@@ -196,24 +197,36 @@ static SdkMeterProvider createMeterProvider(
         shouldSuppressExceptions
             ? new PermissionDeniedSingleReportMetricsExporter(cloudMonitoringExporter)
             : cloudMonitoringExporter;
+    AttributesBuilder attributesBuilder =
+        Attributes.builder()
+            .put("gcp.resource_type", "storage.googleapis.com/Client")
+            .put("project_id", projectIdToUse)
+            .put("instance_id", UUID.randomUUID().toString())
+            .put("api", "grpc");
```

Comment:
nit: could we use something to detect grpc vs http even if this isn't set up for http yet? would make this reusable in the future.

Reply:
AFAIU, resource detector is platform dependent on https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/resourcedetectionprocessor/README.md#gcp-metadata

API type would need to be captured client side / added when introduced into the java storage client.